### PR TITLE
Register with session id

### PR DESCRIPTION
**Pull request by lserra-iov**
Only register pegins with a valid session id.
Always return status ok even when nothing is registered.

### CODE REVIEW

File: src/controllers/register.controller.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  async register(
+    @requestBody({schema: getModelSchemaRef(RegisterPayload)})
+    payload: RegisterPayload,
+  ): Promise<Response> {
+    const {sessionId, type} = payload;
+    let session;
+    if (sessionId) {
+      session = await this.sessionRepository.get(sessionId);
+    }
+    // TODO: remove pegout check
```
Comment (alexjavabraz) [inline src/controllers/register.controller.ts:41]: Remove the comment.
Use the constants to add 'pegout' text.

**Review by alexjavabraz — CHANGES_REQUESTED** [review]
- remote unused comment
- add text into constants.ts file

File: src/controllers/register.controller.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      },
+    },
+  })
+  async register(
+    @requestBody({schema: getModelSchemaRef(RegisterPayload)})
+    payload: RegisterPayload,
+  ): Promise<Response> {
+    const {sessionId, type} = payload;
+    let session;
+    if (sessionId) {
```
Comment (alexjavabraz) [inline src/controllers/register.controller.ts:39]:
```suggestion
        if (sessionId) {
      session = await this.sessionRepository.get(sessionId);
      if (session != null || type === constants.TX_TYPE_PEGOUT) {
        await this.registerService.register(payload);
      }
    }
```

**Review by alexjavabraz — CHANGES_REQUESTED** [review]
avoid re-check when sessionId is null

File: src/controllers/register.controller.ts (hunk 2 above)
Comment (lserra-iov) [reply src/controllers/register.controller.ts:39]: That's not the intended behaviour. If `sessionId` is an empty string (possibly for a peg-out) we don't want to query about the session (it causes a redis error) but we do want to register the peg-out.

**Review by ronaldsg20 — APPROVED** [review]

**Review by alexjavabraz — APPROVED** [review]
LGTM

**Pull request merged**